python/src/python-module.cc

```diff
@@ -0,0 +1,75 @@
+#include <Python.h>
+
+#include "internal/eval.hh"
+#include "internal/ptr.hh"
+
+#include <eval.hh>
+#include <globals.hh>
+#include <shared.hh>
+
+namespace nix::python {
+
+#define _public_ __attribute__((visibility("default")))
+
+PyObject * ThrownNixError = nullptr;
+PyObject * NixError = nullptr;
+
+static PyMethodDef NixMethods[] = {
+    {"eval", (PyCFunction) eval, METH_VARARGS | METH_KEYWORDS, "Eval nix expression"}, {NULL, NULL, 0, NULL}};
+
+static struct PyModuleDef nixmodule = {
+    PyModuleDef_HEAD_INIT, "nix", "Nix expression bindings",
+    -1, /* size of per-interpreter state of the module,
+           or -1 if the module keeps state in global variables. */
+    NixMethods};
+
+extern "C" _public_ PyObject * PyInit_nix(void)
+{
+    // By default, Nix sets the build-hook to be "$(readlink /proc/self/exe) __build-remote", expecting the current
+    // binary to be Nix itself. But when we call the Nix library from Python this isn't the case, the current binary is
+    // Python then So we need to change this default, pointing it to the Nix binary instead
+    nix::settings.buildHook = nix::settings.nixBinDir + "/nix __build-remote";
+    // And by setting buildHook before calling initNix, we can override the defaults without overriding the
```

Comment:
Why do we need the build hook at all? We're just doing instantiations.

Reply:
IFD can start builds.